### DISCUSSION TITLE
Album cover reloads on each song change

### DISCUSSION THREAD

**Discussion by GK-774** [jcorporation/myMPD#715] [General]
myMPD v9.2.2 / mpd 0.23.6 (myMPD shows protocol version 0.23.5)

I´m not sure if this is an actual issue or intentionally so, therefore I´m asking it as a question.

Albums have embedded cover art as well as a folder.jpg in the album directory.
When moving on to the next song of the album the green "no cover" icon appears for the fraction of a second and the cover image is reloaded.
Does myMPD always load the embedded albumart and only fall back to folder.jpg if its not available?
I was under the (probably false) impression that folder.jpg (or cover.jpg) should have precedence.

**Comment by jcorporation** [comment]
myMPD tries the image in the directory first and falls back to embedded image and then tries the albumart and readpicture commands from MPD, but this is handled completely on the serverside, the webclient sees no difference.

**Comment by GK-774** [comment]
Well, the folder.jpg is perfectly readable, is shown and displayed by myMPD when browsing the directory. My question was more if this is queried on each song change (within the same album) and therefore gets reloaded, or if this is not intentional.

**Reply by jcorporation** [reply]
Albumcover is queried each song change for playback view and the small cover in the footer.

How can I reproduce it? Load an album to the queue and looking at the playback card when next song gets played?

**Reply by GK-774** [reply]
Yes, this is how to reproduce it. But from your description I assume this is normal then and nothing to worry about. If the cover is large enough it might well take long enough to just be visible.

**Reply by jcorporation** [reply]
Right, this is by design. However the image is preloaded before fading in and after that the old image is removed. Do you notice differences between browsers?

**Reply by GK-774** [reply]
This was with Firefox. Just tried with Safari on a Mac, same effect. For a very short moment the green card flashes in then the cover is displayed again.

**Reply by GK-774** [reply]
This was with Firefox. Same effect with Safari on a Mac, the green logo flashes in for a short moment and then the cover is displayed again.

**Reply by jcorporation** [reply]
In the playback view? I never noticed this and there should be a fade effect from old to new cover. I check this in my env.

**Comment by GK-774** [comment]
Yes.

**Reply by jcorporation** [reply]
Is this occurring since a specific version?

**Comment by GK-774** [comment]
Hard to say as its very short. I think the first time I have noticed it was about two weeks ago or so, (9.2.2 from git). A fade effect I do not recall to have ever seen.

**Reply by jcorporation** [reply]
I can reproduce this. It is a bug, the fade effect is very short (1s).

**Reply by jcorporation** [reply]
Is now fixed in devel.

**Comment by GK-774** [comment]
Yes, works nicely now, many thanks!

**Comment by GK-774** [comment]
Oh well, now there is another (most likely unintended) effect. With Background Albumart now the opaque cover of the previous album stays and is overlayed with the current cover for the background. I include a screenshot to illustrate it.
![beatles_charlie_parker_overlayed](https://user-images.githubusercontent.com/75694230/160784733-85fea111-ac2d-4fdc-a09c-065263ec0a40.jpg)

**Reply by jcorporation** [reply]
Is now fixed in devel.

**Comment by GK-774** [comment]
Thanks!
